Projects/Contracts/Sources/Viewables/ContractTable.swift

```diff
@@ -76,63 +54,42 @@ extension ContractTable: Viewable {
-                .onValue { contracts in
-                    var contractsToShow = contracts.filter {
-                        switch self.filter {
-                        case .active: return $0.status.asTerminatedStatus == nil
-                        case .terminated: return $0.status.asTerminatedStatus != nil
-                        case .none: return false
-                        }
-                    }
-
-                    if contractsToShow.isEmpty, self.filter.emptyFilter.displaysTerminatedContracts {
-                        contractsToShow = contracts
-                    }
-
-                    let table = Table(
-                        rows: contractsToShow.map { contract -> ContractRow in
-                            ContractRow(
-                                contract: contract,
-                                displayName: contract.displayName,
-                                type: contract.currentAgreement.type,
-                                state: state
-                            )
-                        }
-                    )
-
-                    loadingIndicatorBag.dispose()
-
-                    tableKit.set(table)
+        func getContractsToShow(for state: ContractState) -> [Contract] {
+            switch self.filter {
+            case .active:
+                return state
+                    .contractBundles
+                    .flatMap { $0.contracts }
+            case .terminated:
+                return state.contracts.filter { contract in
+                    contract.currentAgreement?.status == .terminated
                 }
+            case .none: return []
+            }
         }
 
-        watchContracts()
+        bag += store
+            .stateSignal
+            .atOnce()
+            .onValue { state in
+                let contractsToShow = getContractsToShow(for: state)
+
+                let table = Table(
+                    rows: contractsToShow.map { contract -> ContractRow in
+                        ContractRow(
+                            contract: contract
+                        )
+                    }
+                )
 
-        let refreshControl = UIRefreshControl()
-        bag += client.refetchOnRefresh(
-            query: GraphQL.ContractsQuery(locale: Localization.Locale.currentLocale.asGraphQLLocale()),
-            refreshControl: refreshControl
-        )
+                loadingIndicatorBag.dispose()
 
-        tableKit.view.refreshControl = refreshControl
+                tableKit.set(table)
+            }
 
         bag += tableKit.view.didMoveToWindowSignal.onValue { _ in
-            client.fetch(
-                query: GraphQL.ContractsQuery(
-                    locale: Localization.Locale.currentLocale.asGraphQLLocale()
-                ),
-                cachePolicy: .fetchIgnoringCacheData
-            )
-            .onValue { _ in }
+            store.send(.fetchContractBundles)
+            store.send(.fetchContracts)
```

Comment:
Would it makes sense to fetch everything in one big query instead of shooting of two parallel requests?